epaint/src/shape.rs

```diff
@@ -362,6 +378,57 @@ impl From<RectShape> for Shape {
     }
 }
 
+#[derive(Copy, Clone, Debug, PartialEq)]
+#[cfg_attr(feature = "serde", derive(serde::Deserialize, serde::Serialize))]
+/// How rounded the corners of things should be
+pub struct Rounding {
+    pub nw: f32,
+    pub ne: f32,
+    pub sw: f32,
+    pub se: f32,
+}
+
+impl Default for Rounding {
+    fn default() -> Self {
+        Self::none()
+    }
+}
+
+impl From<f32> for Rounding {
+    fn from(radius: f32) -> Self {
+        Self {
+            nw: radius,
+            ne: radius,
+            sw: radius,
+            se: radius,
+        }
+    }
+}
+
+impl Rounding {
+    pub fn new(nw: f32, ne: f32, sw: f32, se: f32) -> Self {
+        Self { nw, ne, sw, se }
+    }
+
```

Comment:
I think this will just lead to confusion - better for users to be explicit imho

```suggestion
```

Reply:
I originally added it with the intention of shortening fmt's tendency to have one field per line for structs regardless of the length, but I'll remove it. I agree it might be confusing to some.